content/blog/custom-docker-images-for-emr-serverless.md

```diff
@@ -0,0 +1,199 @@
+
+<!--StartFragment-->
+
+**docker tag local_docker_image_name:tag *[aws_account_id.dkr.ecr.region.amazonaws.com](http://aws_account_id.dkr.ecr.region.amazonaws.com)\*\****[/docker_image_name:](http://718515174980.dkr.ecr.us-east-1.amazonaws.com/emr-serverless-ci-examples:emr-serverless-6.9.0-V1)tag\*\*
+
+<!--EndFragment-->
+
+Command to authenticate Docker to an AWS ECR private registry assuming region as us-east-1. Run below command before pushing the docker image to AWS ECR private registry.
+
+<!--StartFragment-->
+
+***aws ecr get-login-password — region us-east-1 — profile <profile_name_in_your_aws_credentials_file> | docker login — username AWS — password-stdin [aws_account_id.dkr.ecr.region.amazonaws.com](http://aws_account_id.dkr.ecr.region.amazonaws.com)***
+
+<!--EndFragment-->
+
+        **Note:**
+
+        1. When --profile option is not provided, then credentials will be picked from profile name <default> in the \~/.aws/credentials file. If you are using access credentials for different user, then include the profile section in \~/.aws/credentials files.
+
+        2.  Example content of ~/.aws/credentials file:          
+
+<!--StartFragment-->
+
+\[default]\
+aws_access_key_id =\
+aws_secret_access_key =
+
+\[testprofile]\
+aws_access_key_id=\
+aws_secret_access_key=
+
+<!--EndFragment-->
+
+Command to push docker image to AWS ECR private registry:
+
+       **docker push *[aws_account_id.dkr.ecr.region.amazonaws.com](http://aws_account_id.dkr.ecr.region.amazonaws.com)\*\****[/docker_image_name:](http://718515174980.dkr.ecr.us-east-1.amazonaws.com/emr-serverless-ci-examples:emr-serverless-6.9.0-V1)tag\*\*
+
+## **Step 4:  EMR Serverless using the custom Docker Image Created.**
+
+There are two approaches for EMR Serverless to use the custom Image created
+
+(a) via AWS CLA
+
+(b) via EMR serverless application settings in AWS management console
+
+### **4a) Via AWS CLI command to create simple EMRServerless application using custom docker image:**
+
+AWS CLI reference for EMRServerless application management: <https://docs.aws.amazon.com/cli/latest/reference/emr-serverless/index.html>
+
+<!--StartFragment-->
+
+aws — region <region> emr-serverless create-application \
+— release-label emr-6.9.0 \
+— type “SPARK” \
+— name emr-application-1 \
+**— image-configuration ‘{ “imageUri”: “<your AWS account ID>.dkr.ecr.<region>.[amazonaws.com/<ecr_registry_name:](http://amazonaws.com/emr-serverless-ci-examples:emr-serverless-ci-ml)image_name>” }’**
+
+<!--EndFragment-->
+
+**Note:** Image URI can be copied from AWS ECR registry.
+
+### **4b) Via EMR Serverless Settings in the EMR Management console**
+
+##### **EMRServerless Custom Image Settings section in AWS EMR Serverless management console:**
+
+![EMR Serverless Management Console Docker section](/img/customimagesettings-in-emrserverless.png)
+
+##### Browse and select custom EMRServerless image from AWS ECR private registry in the same region:
+
```

Comment:
Before you get into the references and About the authors sections, I'd recommend summarizing the article for the reader - giving him/her a brief synopsis of what you just went through and why, mentioning that you hope the reader found it helpful, and giving them a call to action - which might perhaps be to read the documentation below. For example "If you found this blog post helpful, we recommend you read more on the topic by referencing the documentation below:" (or something similar...)